### PR TITLE
Small integer preallocation optimization 

### PR DESCRIPTION
**Pull request by patiences**
This PR implements the preallocation and initialization of the integers [-5, 257) so that they can be reused. In general, this means that when the python code is parsed and an integer object is detected, if it is in this range we will not create a new Integer object but rather use the preallocated one. 

Object construction is expensive memory and time-wise. We can see a performance gain even running a simple performance test like [this one](https://github.com/pybee/voc/pull/824/commits/a89fc6617b582c3b5981df88fb4a2d9c2376edeb): 

Pre-optimization 
```
Running test_small_integers ...
  Elapsed time:  33.07889499713201  ms
  CPU process time:  289.66  ms
```
With optimization 
```
Running test_small_integers ...
  Elapsed time:  23.559533001796808  ms
  CPU process time:  240.0899999999998  ms
```
(Of course there is always some variance between different runs, but from what I've seen there is consistent improvement.)

### CODE REVIEW

**Comment by patiences** [comment]
As a further piece of work I think we should consider restricting access to the `Int` constructor altogether. This is perhaps a small issue on the transpilation side, for example, when we have an instruction that creates an integer from a value inside the variable so we're not able to check if the integer is small and generate lookup instructions. However, the math operations on `Int` objects create new `Int` objects all the time. It should be fairly straightforward to refactor the code to check for small integers, if it's something we want to do.

**Comment by patiences** [comment]
Some updated numbers: 

Running test_small_integers
  Elapsed time:  122.01105412701145  sec
  CPU process time:  0.0006999999999999784  sec

Running test_small_integers
  Elapsed time:  128.66034104398568  sec
  CPU process time:  0.0008010000000000517  sec

Running test_small_integers
  Elapsed time:  122.50824150099652  sec
  CPU process time:  0.00048599999999998644  sec

Running test_small_integers
  Elapsed time:  122.89947666198714  sec
  CPU process time:  0.0008429999999999827  sec

Running test_small_integers
  Elapsed time:  122.97956180298934  sec
  CPU process time:  0.00045000000000000595  sec

**Comment by freakboy3742** [comment]
@patiences Do you have baseline numbers for the benchmark? The ticket only has the "progress" numbers...

**Comment by patiences** [comment]
@freakboy3742  Yes, of course! Here we go: 

**Pre-optimization** 
Running test_small_integers
  Elapsed time:  144.74749750000774  sec
  CPU process time:  0.00047200000000002795  sec

Running test_small_integers
  Elapsed time:  143.55906505999155  sec
  CPU process time:  0.0010949999999999571  sec

Running test_small_integers
  Elapsed time:  149.2436827890051  sec
  CPU process time:  0.0008159999999999834  sec

Running test_small_integers
  Elapsed time:  146.150813241984  sec
  CPU process time:  0.0008259999999999934  sec

Looks like about a 15% improvement.

**Comment by patiences** [comment]
> python/testfile.class has also been inadvertently added.

>That file needs to be stripped out (ideally, rolling back the commit where it was added; if you just >delete the file, it will still be in the commit history, and so every person who clones the repo will >download the file, even though it doesn't exist any more).

Really? That is so weird. Is that the case with .py files too (some of those got added and deleted at other points too, oops)? I actually am not sure how to revert just the specific commit (bb40b5b) -- I tried to rebase it without that commit locally but I'm not sure that's going to work...

File: voc/python/blocks.py [hunk 1]
```diff
@@ -16,6 +16,10 @@
 
 # from .debug import DEBUG, DEBUG_value
 
+# The number of negative and positive small org/python/types/Int objects that are preallocated
+# [-NSMALLNEGINTS, NSMALLPOSINTS)
+NSMALLNEGINTS = 5
+NSMALLPOSINTS = 257
```
Comment (freakboy3742) [inline voc/python/blocks.py]: I don't think this is needed anymore

**Pull request merged**